drivers/SmartThings/zigbee-switch/src/inovelli-vzm31-sn/init.lua

```diff
@@ -0,0 +1,414 @@
+          device:emit_component_event(
+            component,
+            capabilities.button.supportedButtonValues(
+              {"pushed","held","down_hold","pushed_2x","pushed_3x","pushed_4x","pushed_5x"},
+              { visibility = { displayed = false } }
+            )
+          )
+        end
+        if device:get_latest_state(component.id, capabilities.button.ID, capabilities.button.numberOfButtons.NAME) == nil then
+          device:emit_component_event(
+            component,
+            capabilities.button.numberOfButtons({value = 1}, { visibility = { displayed = false } })
+          )
+        end
+      end
+    end
+    device:send(cluster_base.read_attribute(device, data_types.ClusterId(0x0000), 0x4000))
+  else
+    device:emit_event(capabilities.colorControl.hue(1))
+    device:emit_event(capabilities.colorControl.saturation(1))
+    device:emit_event(capabilities.colorTemperature.colorTemperature(6500))
+    device:emit_event(capabilities.switchLevel.level(100))
+    device:emit_event(capabilities.switch.switch("off"))
+  end
+end
+
+local function energy_meter_handler(driver, device, value, zb_rx)
+  local raw_value = value.value
+  raw_value = raw_value / 100
+  device:emit_event(capabilities.energyMeter.energy({value = raw_value, unit = "kWh" }))
+end
+
+local function power_meter_handler(driver, device, value, zb_rx)
+  local raw_value = value.value
+  raw_value = raw_value / 10
+  device:emit_event(capabilities.powerMeter.power({value = raw_value, unit = "W" }))
+end
+
+local function huePercentToValue(value)
+  if value <= 2 then
+    return 0
+  elseif value >= 98 then
+    return 255
+  else
+    return utils.round(value / 100 * 255)
+  end
+end
+
+local function getNotificationValue(device, value)
+  local notificationValue = 0
+  local level = device:get_latest_state("main", capabilities.switchLevel.ID, capabilities.switchLevel.level.NAME) or 100
+  local color = utils.round(device:get_latest_state("main", capabilities.colorControl.ID, capabilities.colorControl.hue.NAME) or 100)
+  local effect = device:get_parent_device().preferences.notificationType or 1
+  notificationValue = notificationValue + (effect*16777216)
+  notificationValue = notificationValue + (huePercentToValue(value or color)*65536)
+  notificationValue = notificationValue + (level*256)
+  notificationValue = notificationValue + (255*1)
+  return notificationValue
+end
+
+local function on_handler(driver, device, command)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD then
+    device:send(clusters.OnOff.server.commands.On(device))
+  else
+    device:emit_event(capabilities.switch.switch("on"))
+    local dev = device:get_parent_device()
+    local send_configuration = function()
+      dev:send(cluster_base.build_manufacturer_specific_command(
+            dev,
+            0xfc31,
+            0x01,
+            0x122f,
```

Comment:
can you pull the cluster_id and mfg_code out and put them at the top of the file?